### PR TITLE
Update version of GitHub Actions

### PR DESCRIPTION
**Pull request by git-hulk**
This PR updates the version of the following actions:

- hecrj/setup-rust-action from v1 to v2
- actions/checkout from v2 to v4

### CODE REVIEW

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -8,31 +8,31 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Set up Rust
-        uses: hecrj/setup-rust-action@v1
+        uses: hecrj/setup-rust-action@v2
```
Comment (alamb) [inline .github/workflows/rust.yml:11]: I am a little worried about this action as it doesn't look very actively maintained 

https://github.com/hecrj/setup-rust-action/issues

However, this is probably ok -- maybe we can find time to setup rust ourselves (using `rustup install`)

Here is how we do it in DataFusion: https://github.com/apache/datafusion/blob/main/.github/actions/setup-builder/action.yaml

**Review by alamb — APPROVED** [review]
Thanks @iffyio  and @git-hulk

File: .github/workflows/rust.yml (hunk 1 above)
Comment (git-hulk) [reply .github/workflows/rust.yml:11]: @alamb Thanks for your review and comment. I will be happy to do this when I get time.

**Pull request merged**